rspace/src/test/scala/coop/rchain/rspace/test/ArbitraryInstances.scala

```diff
@@ -84,6 +84,24 @@ object ArbitraryInstances {
         .map(_.toMap))
   }
 
+  implicit val arbitraryWaitingContinuation
```

Comment:
Could you refactor `arbitraryNonEmptyMapListStringWaitingContinuation` to use this?

(Also: great name, right? 😛 )

Reply:
Sure thing!

(Agreed, the name is a bit too descriptive ;-) )

Reply:
So I did a poor job of expressing this in my last comment, but I would like to explicitly maintain the invariant of the channels length always matching the patterns length in the `Arbitrary` instances themselves.  This requires a little bit more rejiggering:

```scala
  def arbitraryWaitingContinuation(
      patternsLength: Int): Arbitrary[WaitingContinuation[Pattern, StringsCaptor]] =
    Arbitrary(for {
      pats    <- Gen.containerOfN[List, Pattern](patternsLength, Arbitrary.arbitrary[Pattern])
      boolean <- Arbitrary.arbitrary[Boolean]
    } yield WaitingContinuation(pats, new StringsCaptor, boolean))

  implicit val arbitraryGnat: Arbitrary[GNAT[String, Pattern, String, StringsCaptor]] = {
    Arbitrary(Gen.sized { size =>
      val constrainedSize = if (size > 1) size else 1
      for {
        chans <- Gen.containerOfN[List, String](constrainedSize, Arbitrary.arbitrary[String])
        data  <- Gen.containerOf[List, Datum[String]](Arbitrary.arbitrary[Datum[String]])
        wks <- Gen.containerOf[List, WaitingContinuation[Pattern, StringsCaptor]](
                arbitraryWaitingContinuation(constrainedSize).arbitrary)
      } yield GNAT(chans, data, wks)
    })
  }

  implicit val arbitraryNonEmptyMapListStringWaitingContinuation
    : Arbitrary[Map[List[String], WaitingContinuation[Pattern, StringsCaptor]]] = {
    Arbitrary(Gen.sized { size =>
      val constrainedSize = if (size > 1) size else 1
      Gen
        .nonEmptyContainerOf[List, (List[String], WaitingContinuation[Pattern, StringsCaptor])](
          for {
            wc    <- arbitraryWaitingContinuation(constrainedSize).arbitrary
            chans <- Gen.containerOfN[List, String](constrainedSize, Arbitrary.arbitrary[String])
          } yield (chans, wc)
        )
        .map(_.toMap)
    })
  }
```

Reply:
Correction:
```scala
  def arbitraryWaitingContinuation(
      patternsLength: Int): Arbitrary[WaitingContinuation[Pattern, StringsCaptor]] =
    Arbitrary(Gen.sized { _ =>
      for {
        pats    <- Gen.containerOfN[List, Pattern](patternsLength, Arbitrary.arbitrary[Pattern])
        boolean <- Arbitrary.arbitrary[Boolean]
      } yield WaitingContinuation(pats, new StringsCaptor, boolean)
    })

  implicit val arbitraryGnat: Arbitrary[GNAT[String, Pattern, String, StringsCaptor]] = {
    Arbitrary(Gen.sized { size =>
      val constrainedSize = if (size > 1) size else 1
      for {
        chans <- Gen.containerOfN[List, String](constrainedSize, Arbitrary.arbitrary[String])
        data  <- Gen.nonEmptyContainerOf[List, Datum[String]](Arbitrary.arbitrary[Datum[String]])
        wks <- Gen.nonEmptyContainerOf[List, WaitingContinuation[Pattern, StringsCaptor]](
                arbitraryWaitingContinuation(constrainedSize).arbitrary)
      } yield GNAT(chans, data, wks)
    })
  }

  implicit val arbitraryNonEmptyMapListStringWaitingContinuation
    : Arbitrary[Map[List[String], WaitingContinuation[Pattern, StringsCaptor]]] = {
    Arbitrary(Gen.sized { size =>
      val constrainedSize = if (size > 1) size else 1
      Gen
        .nonEmptyContainerOf[List, (List[String], WaitingContinuation[Pattern, StringsCaptor])](
          for {
            wc    <- arbitraryWaitingContinuation(constrainedSize).arbitrary
            chans <- Gen.containerOfN[List, String](wc.patterns.length, Arbitrary.arbitrary[String])
          } yield (chans, wc)
        )
        .map(_.toMap)
    })
  }
```